From e817decdab489f02506749c3818fb8c2150c9862 Mon Sep 17 00:00:00 2001
From: Avinash Kumar <avikumar@habana.ai>
Date: Thu, 11 Nov 2021 03:31:22 +0200
Subject: [PATCH] [SW-62043] greco: fit image with with sram

Relocation of some of greco mgmt drivers
from iccm to sram needed to save iccm. Fit
image creation related changes done to scripts.

Signed-off-by: Avinash Kumar <avikumar@habana.ai>
Change-Id: I9426400338c345754b1653743ec59cfe5b5cf829
---

diff --git a/board/habanalabs/common/post-image-fit-global.sh b/board/habanalabs/common/post-image-fit-global.sh
index 5733ada..e66efcd 100755
--- a/board/habanalabs/common/post-image-fit-global.sh
+++ b/board/habanalabs/common/post-image-fit-global.sh
@@ -66,6 +66,10 @@
 
 ITS_SED_PARAMS+="s/@DEV_PREFIX@/${DEV_PREFIX}/g;"
 ITS_SED_PARAMS+="s/@FW_LOADER_CONF_IMAGES@/${FW_LOADER_CONF_IMAGES}/g;"
+ITS_SED_PARAMS+="s/@PREBOOT_CONF_IMAGES@/${PREBOOT_CONF_IMAGES}/g;"
+ITS_SED_PARAMS+="s/@PREBOOT_SIGN_IMAGES@/${PREBOOT_SIGN_IMAGES}/g;"
+ITS_SED_PARAMS+="s/@ZEPHYR_MGMT_SRAM_IMAGE@/${ZEPHYR_MGMT_SRAM_IMAGE}/g;"
+
 ITS_SED_PARAMS+="s/@HL_ARCH@/${HL_ARCH}/g;"
 
 KEY_NAME=hl_rsa3072
@@ -87,6 +91,8 @@
 ITS_SED_PARAMS+="s/@ARC_MGMT_DCCM_LO@/${ARC_MGMT_DCCM_LO}/g;"
 ITS_SED_PARAMS+="s/@ARC_MGMT_ICCM_HI@/${ARC_MGMT_ICCM_HI}/g;"
 ITS_SED_PARAMS+="s/@ARC_MGMT_ICCM_LO@/${ARC_MGMT_ICCM_LO}/g;"
+ITS_SED_PARAMS+="s/@ARC_MGMT_SRAM_HI@/${ARC_MGMT_SRAM_HI}/g;"
+ITS_SED_PARAMS+="s/@ARC_MGMT_SRAM_LO@/${ARC_MGMT_SRAM_LO}/g;"
 ITS_SED_PARAMS+="s/@ARC_PID_DCCM_HI@/${ARC_PID_DCCM_HI}/g;"
 ITS_SED_PARAMS+="s/@ARC_PID_DCCM_LO@/${ARC_PID_DCCM_LO}/g;"
 ITS_SED_PARAMS+="s/@ARC_PID_ICCM_HI@/${ARC_PID_ICCM_HI}/g;"
diff --git a/board/habanalabs/common/uboot-zephyr.its b/board/habanalabs/common/uboot-zephyr.its
index b7688ff..34c5d21 100644
--- a/board/habanalabs/common/uboot-zephyr.its
+++ b/board/habanalabs/common/uboot-zephyr.its
@@ -197,6 +197,19 @@
 				algo = "@SHA_VERSION@";
 			};
 		};
+
+		zephyr-mgmt-sram {
+			description = "Zephyr mgmt SRAM";
+			data = /incbin/("@HL_DEV@-zephyr-mgmt-sram.bin.le");
+			type = "standalone";
+			arch = "arc";
+			compression = "none";
+			loadaddr = <@ARC_MGMT_SRAM_HI@ @ARC_MGMT_SRAM_LO@>;
+			hash {
+				algo = "@SHA_VERSION@";
+			};
+		};
+
 	};
 
 	configurations {
@@ -210,21 +223,18 @@
 
 		preboot-conf {
 			description = "Habanalabs device preboot";
-			loadables = "uboot", "zephyr-pid-iccm",
-				    "zephyr-pid-dccm", "zephyr-mgmt-iccm",
-				    "zephyr-mgmt-dccm";
+			loadables = @PREBOOT_CONF_IMAGES@;
 			uboot = "uboot";
 			zephyr-pid-iccm = "zephyr-pid-iccm";
 			zephyr-pid-dccm = "zephyr-pid-dccm";
 			zephyr-mgmt-iccm = "zephyr-mgmt-iccm";
 			zephyr-mgmt-dccm = "zephyr-mgmt-dccm";
+			zephyr-mgmt-sram = @ZEPHYR_MGMT_SRAM_IMAGE@;
 			nic_fw = "nic_fw";
 			signature {
 				algo = "@SHA_VERSION@,rsa3072";
 				key-name-hint = "hl_rsa3072";
-				sign-images = "uboot", "nic_fw", "zephyr-pid-iccm",
-					"zephyr-pid-dccm", "zephyr-mgmt-iccm",
-					"zephyr-mgmt-dccm";
+				sign-images = @PREBOOT_SIGN_IMAGES@;
 			};
 		};
 	};
diff --git a/board/habanalabs/gaudi2/post-image-fit.sh b/board/habanalabs/gaudi2/post-image-fit.sh
index 164d470..e2abcb1 100644
--- a/board/habanalabs/gaudi2/post-image-fit.sh
+++ b/board/habanalabs/gaudi2/post-image-fit.sh
@@ -20,6 +20,8 @@
 ARC_MGMT_DCCM_LO=0xF7060000
 ARC_MGMT_ICCM_HI=0x1000007F
 ARC_MGMT_ICCM_LO=0xF7040000
+ARC_MGMT_SRAM_HI=0x1000007F
+ARC_MGMT_SRAM_LO=0xF7FE0000
 ARC_PID_DCCM_HI=0x1000007F
 # Should be aligned with arcpid DCCM_START
 ARC_PID_DCCM_LO=0xF7021400
@@ -32,6 +34,10 @@
 PPBOOT_PRI_OFFSET=0x00030000
 PPBOOT_SEC_OFFSET=0x00050000
 FW_LOADER_CONF_IMAGES=`echo '"security", "preboot", "ppboot", "img-ptr"'`
+PREBOOT_CONF_IMAGES=`echo '"uboot", "zephyr-pid-iccm", "zephyr-pid-dccm", "zephyr-mgmt-iccm", "zephyr-mgmt-dccm"'`
+PREBOOT_SIGN_IMAGES=`echo '"uboot", "nic_fw", "zephyr-pid-iccm", "zephyr-pid-dccm", "zephyr-mgmt-iccm", "zephyr-mgmt-dccm"'`
+# Gaudi2 no code relocation so its empty
+ZEPHYR_MGMT_SRAM_IMAGE=`echo '""'`
 HL_ARCH="arc"
 # Set hl-preboot-version = "hl-<device>-<HL_VERSION>[-rc]-fw-<fw ver>[-<git data>] (<date> - <time>)"
 HL_PREBOOT_VERSION=`cat ${BR2_EXTERNAL}/output/${DEV_NAME}_${HL_ARCH}/build/preboot-custom/buildversion-habanalabs`
@@ -56,6 +62,4 @@
 	cp ${BR2_EXTERNAL}/output/${DEV_NAME}_arc/images/${DEV_PREFIX}-zephyr-mgmt-iccm.bin ${FILEPATH}/${DEV_PREFIX}-zephyr-mgmt-iccm.bin.le
 	cp ${BR2_EXTERNAL}/output/${DEV_NAME}_arc/images/${DEV_PREFIX}-zephyr-mgmt-dccm.bin ${FILEPATH}/${DEV_PREFIX}-zephyr-mgmt-dccm.bin.le
 	cp ${BR2_EXTERNAL}/output/${DEV_NAME}_arc/images/${DEV_PREFIX}-zephyr-mgmt-sram.bin ${FILEPATH}/${DEV_PREFIX}-zephyr-mgmt-sram.bin.le
-
-	return
 }
diff --git a/board/habanalabs/greco/post-image-fit.sh b/board/habanalabs/greco/post-image-fit.sh
index d5d4b34..ebe2996 100755
--- a/board/habanalabs/greco/post-image-fit.sh
+++ b/board/habanalabs/greco/post-image-fit.sh
@@ -15,6 +15,8 @@
 ARC_MGMT_DCCM_LO=0xFB030000
 ARC_MGMT_ICCM_HI=0x0000007F
 ARC_MGMT_ICCM_LO=0xFB020000
+ARC_MGMT_SRAM_HI=0x0000007F
+ARC_MGMT_SRAM_LO=0xFBFE0000
 ARC_PID_DCCM_HI=0x0000007F
 ARC_PID_DCCM_LO=0xFB010C00
 ARC_PID_ICCM_HI=0x0000007F
@@ -27,6 +29,9 @@
 PPBOOT_SEC_OFFSET=0x00050000
 
 FW_LOADER_CONF_IMAGES=`echo '"btl-s1", "btl-s2", "ppboot", "img-ptr", "preboot"'`
+PREBOOT_CONF_IMAGES=`echo '"uboot", "zephyr-pid-iccm", "zephyr-pid-dccm", "zephyr-mgmt-iccm", "zephyr-mgmt-dccm", "zephyr-mgmt-sram"'`
+PREBOOT_SIGN_IMAGES=`echo '"uboot", "nic_fw", "zephyr-pid-iccm", "zephyr-pid-dccm", "zephyr-mgmt-iccm", "zephyr-mgmt-dccm", "zephyr-mgmt-sram"'`
+ZEPHYR_MGMT_SRAM_IMAGE=`echo '"zephyr-mgmt-sram"'`
 HL_ARCH="arc"
 # Set hl-preboot-version = "hl-<device>-<HL_VERSION>[-rc]-fw-<fw ver>[-<git data>] (<date> - <time>)"
 HL_PREBOOT_VERSION=`cat ${BR2_EXTERNAL}/output/${DEV_NAME}_arc/build/preboot-custom/buildversion-habanalabs`
